baselayout/nsswitch.conf

```
@@ -4,7 +4,7 @@ passwd:      files usrfiles sss systemd
 shadow:      files usrfiles sss
 group:       files usrfiles sss systemd
 
-hosts:       files usrfiles dns myhostname
+hosts:       resolve [!UNAVAIL=return] files usrfiles dns myhostname
```

The first to look up should still be files and usrfiles:
```suggestion
hosts:       files usrfiles resolve [!UNAVAIL=return] dns myhostname
```

Actually I find it strange that `myhostname` comes last because this is normally between `files (usrfiles)` and `resolve` but that's not part of this PR.

---

I took https://www.freedesktop.org/software/systemd/man/nss-systemd.html as example and moved myhostname to the back as it was already the last.

The ordering with files first makes little sense as resolved will already include those in the lookup which is why it's suggested to be first and have a quick return

---

I think the situation with `usrfiles` (nss-altfiles) may not be covered by `resolve`, that's why I suggested to move it just before `dns`.

---

I was looking into usrfiles (https://github.com/kubic-project/libnss_usrfiles ?) and it doesnt seem to care about hosts at all and only deals with user/group... In which case we can scrap it and put resolve back to the front to enable files caching

---

It's this here https://github.com/kinvolk/nss-altfiles and supports looking up in `/usr/share/baselayout/hosts`.

---

Okay but according to that documentation it should be `altfiles` and not `usrfiles`

---

Yes, a bit confusing, it's compiled with the option `--with-module-name=usrfiles`.

---

Ah ok, yeah slightly confusing ;)

I guess I mark this as resolved again 👍

---

Good, I'm running a test with all changes combined, thanks again for the PRs

---

There are test failures on Azure (the rest is fine except for silencing a warning about a new listening socket that is detected), I'll have to reproduce them and see what's wrong.

---

Which failures? happy to help

The new socket is to be expected due to the stub listener being enabled now

---

Eventually it works but systemd-resolved tries DNSSEC first and that fails.

---

Define fails, DNSSEC is preferred but defaults to allow-downgrade which should transparently retry without DNSSEC. Queries should only fail if your recursive DNS server sends bogus DNSSEC responses at which point the local resolver has to assume that it's a forged response and drop the answer accordingly.

---

It failed to resolve because DNSSEC is not supported and the retry was not transparent but failed the whole resolving action. I think DNSSEC should be opt-in until this behavior gets better.